### PR TITLE
Authorization: Add READ_ABOUT privilege for being able to see information about space; remove anonymousReadAccess

### PR DESCRIPTION
**Pull request by techsmyth**
added new privilege: READ_ABOUT

Tidied up credential rules related to space admins, removing one rule

Added in credential rule to assign READ_ABOUT to all members of parent space for Space itself and also for Context + Profile

Added privilege rule so that if a user has READ then also gets READ_ABOUT, for Space, Context + Profile

Updated field resolvers on Space to check for READ_ABOUT (previously were simply left without authorization checks)

Updated lookup of context, profiles + spaces to check for READ_ABOUT

Triggered by work on READ_ABOUT...where we will need more control that the simple flag that controls READ access.

=======
Add new global credential GLOBAL_ANONYMOUS
Assign this credential to AgentInfos that are representing anonymous usage

Removed anonymousReadAccess flag from authorization policy
Added utility method to assign a credential rule for anonymous usage with a specified privilege to an authorization policy

Updated Space sorting to use privacy mode (public/private) instead of anonymousReadAccess (basically fixing a potential bug). Potentially the new setup is also faster.

Reworked handling of AgentInfo to bring the creation / updating of this data structure to be in one place

Client is already updated not to use anonymousReadAccess flag. 

Initial results:
image

Todo:
- not yet tested / run
- check what updates the client needs
- check for edge cases by having test suites passing; there may be cases where something is over ruled in the child previously which the new setup will not cover
- update unit tests on Space: they were using anonymousReadAccess but now should be using settings.privacy.mode = Public/Private
- see if can easily share the logic for generating an AgentInfo for anonymous user requests, currently duplicated 3x. Problems with module dependency in one case, plus second case where have functional vs service usage.
- check with light testing the impact this has on performance; risk is that it is heavier
- test test test

## Release Notes

- **New Features**
  - Added new authorization privileges and policy rules for reading about entities.
  - Introduced a new constant for credential rules related to space members.

- **Authorization Changes**
  - Removed `anonymousReadAccess` property from authorization policies across various entities.
  - Enhanced credential-based access control mechanisms.
  - Implemented more granular privilege mapping for read access.

- **Authentication**
  - Updated agent information handling to support anonymous and authenticated users.
  - Refined service methods for managing user and agent credentials.

- **Refactoring**
  - Replaced direct anonymous access assignments with credential rule methods.
  - Updated type definitions for credentials across multiple services.
  - Simplified authorization logic in various modules.

- **Compatibility**
  - Migrated existing authorization policies to new credential-based approach.
  - Removed legacy anonymous access configurations.

### CODE REVIEW

**Comment by valentinyanakiev** [comment]
> added new privilege: READ_ABOUT
> 
> Tidied up credential rules related to space admins, removing one rule
> 
> Added in credential rule to assign READ_ABOUT to all members of parent space for Space itself and also for Context + Profile
> 
> Added privilege rule so that if a user has READ then also gets READ_ABOUT, for Space, Context + Profile
> 
> Updated field resolvers on Space to check for READ_ABOUT (previously were simply left without authorization checks)
> 
> Updated lookup of context, profiles + spaces to check for READ_ABOUT
> 
> To do:
> 
> * not yet tested / run
> * check what updates the client needs

I have different design / considerations that in my opinion is better / will require less changes:
https://alkem.io/building-alkemio-org/challenges/technicaldesigns-9853/collaboration/readingaboutprivileges

It also follows the pattern used in other places where we assign a specific credential to the parent and then we have a privilege rule to 'translate' it to the child.

File: src/domain/space/space/space.service.authorization.ts [hunk 1]
```diff
@@ -346,42 +392,52 @@ export class SpaceAuthorizationService {
...
+      if (parentRoleSetMemberCredentials.length !== 0) {
+        const readAboutSubspaces =
+          this.authorizationPolicyService.createCredentialRule(
+            [AuthorizationPrivilege.DELETE],
+            parentRoleSetAdminCredentials,
+            CREDENTIAL_RULE_SPACE_ADMIN_DELETE_SUBSPACE
+          );
+        readAboutSubspaces.cascade = false;
+        newRules.push(readAboutSubspaces);
+      }
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply src/domain/space/space/space.service.authorization.ts]: good catch

File: src/services/external/excalidraw-backend/middlewares/socket.data.init.middleware.ts [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
   socket.data.lastContributed = -1;
   socket.data.read = false;
   socket.data.update = false;
   socket.data.session = undefined;
 
   next();
 };
+
+// Todo: duplicate as have both service + function implementations
+const createAnonymousAgentInfo = (): AgentInfo => {
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply src/services/external/excalidraw-backend/middlewares/socket.data.init.middleware.ts]: SimpleMiddlewareHandler accepts only 2 parameters, export type SimpleMiddlewareHandler = (
  socket: SocketIoSocket,
  next: (err?: ExtendedError) => void
) => Awaited<void>;

**Review by valentinyanakiev — APPROVED** [review]
